Known Indel tracker is non-serializable

Indel realignment fails with a serialization error if you run it with a file containing known indels:

```
Exception in thread "main" org.apache.spark.SparkException: Task not serializable
        at org.apache.spark.util.ClosureCleaner$.ensureSerializable(ClosureCleaner.scala:166)
        at org.apache.spark.util.ClosureCleaner$.clean(ClosureCleaner.scala:158)
        at org.apache.spark.SparkContext.clean(SparkContext.scala:1435)
        at org.apache.spark.rdd.RDD.flatMap(RDD.scala:278)
        at org.bdgenomics.adam.rdd.read.realignment.RealignIndels.realignIndels(RealignIndels.scala:495)
        at org.bdgenomics.adam.rdd.read.realignment.RealignIndels$.apply(RealignIndels.scala:58)
        at org.bdgenomics.adam.rdd.read.AlignmentRecordRDDFunctions$$anonfun$adamRealignIndels$1.apply(AlignmentRecordRDDFu
nctions.scala:327)
        at org.bdgenomics.adam.rdd.read.AlignmentRecordRDDFunctions$$anonfun$adamRealignIndels$1.apply(AlignmentRecordRDDFu
nctions.scala:327)
        at org.apache.spark.rdd.Timer.time(Timer.scala:57)
        at org.bdgenomics.adam.rdd.read.AlignmentRecordRDDFunctions.adamRealignIndels(AlignmentRecordRDDFunctions.scala:326
)
        at org.bdgenomics.adam.cli.Transform.apply(Transform.scala:110)
        at org.bdgenomics.adam.cli.Transform.run(Transform.scala:139)
        at org.bdgenomics.utils.cli.BDGSparkCommand$class.run(BDGCommand.scala:53)
        at org.bdgenomics.adam.cli.Transform.run(Transform.scala:86)
        at org.bdgenomics.adam.cli.ADAMMain$.main(ADAMMain.scala:105)
        at org.bdgenomics.adam.cli.ADAMMain.main(ADAMMain.scala)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.apache.spark.deploy.SparkSubmit$.launch(SparkSubmit.scala:358)
        at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:75)
        at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
Caused by: java.io.NotSerializableException: org.apache.spark.SparkContext
        at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1183)
        at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1547)
        at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1508)
        at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1431)
        at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java
```

This was reported on gitter by @jstjohn.

--- a/adam-core/src/main/scala/org/bdgenomics/adam/algorithms/consensus/ConsensusGeneratorFromKnowns.scala
+++ b/adam-core/src/main/scala/org/bdgenomics/adam/algorithms/consensus/ConsensusGeneratorFromKnowns.scala
@@ -24,8 +24,9 @@ import org.bdgenomics.adam.rdd.ADAMContext._
 import org.bdgenomics.adam.rdd.read.realignment.IndelRealignmentTarget
 import org.bdgenomics.adam.rich.RichAlignmentRecord
 import org.bdgenomics.formats.avro.Variant
+import scala.transient
 
-class ConsensusGeneratorFromKnowns(file: String, sc: SparkContext) extends ConsensusGenerator {
+class ConsensusGeneratorFromKnowns(file: String, @transient sc: SparkContext) extends ConsensusGenerator {
 
   val indelTable = sc.broadcast(IndelTable(file, sc))
 

--- a/adam-core/src/main/scala/org/bdgenomics/adam/models/Consensus.scala
+++ b/adam-core/src/main/scala/org/bdgenomics/adam/models/Consensus.scala
@@ -20,7 +20,7 @@ package org.bdgenomics.adam.models
 import htsjdk.samtools.{ Cigar, CigarOperator }
 import org.bdgenomics.adam.util.ImplicitJavaConversions._
 
-object Consensus {
+object Consensus extends Serializable {
 
   def generateAlternateConsensus(sequence: String, start: ReferencePosition, cigar: Cigar): Option[Consensus] = {
 

